soup.go

```diff
@@ -293,6 +293,17 @@ checkNode:
 	return ""
 }
 
+// HTML returns the HTML code for the specific element
+
+func (r Root) HTML() string {
+	var buf bytes.Buffer
+	e := html.Render(&buf, r.Pointer)
+	if e != nil {
+		return ""
+	}
```

Comment:
In Go these preceding lines are normally written using the following idiom:
```
if err := html.Render(&bug, r.Pointer); err != nil {
	return ""
}
```
This accomplishes two things:

1. It makes it clear that the variable (`err`) is an error and that we are handling the error right away.
2. The `err` variable is scoped to the `if` block so it does not leak out into other logic that follows.

Reply:
+1